consul/kv_consul.go

```diff
@@ -141,6 +126,40 @@ func New(
 	}, nil
 }
 
+// New constructs a new kvdb.Kvdb given a list of end points to conntect to.
+func New(
+	domain string,
+	machines []string,
+	options map[string]string,
+	fatalErrorCb kvdb.FatalErrorCB,
+) (kvdb.Kvdb, error) {
+	var kv kvdb.Kvdb
+	var err error
+	if len(machines) == 0 {
+		for _, machine := range defaultMachines {
+			machine := machine
+			if kv, err = newKv(domain, machine, options, fatalErrorCb); err == nil {
+				// return on success, otherwise keep trying next end points
+				return kv, nil
+			}
+		}
+	} else {
+		for _, machine := range machines {
+			machine := machine
```

Comment:
not required ?

Reply:
in this case not required, but i always prefer putting it in whenever we pass range values to func